### ISSUE TITLE
case statements over enums should not require/allow a default statement

### ISSUE THREAD

**Issue by kasimon** [rodjek/puppet-lint#865]
Given a variable defined like

    Enum['up','down'] $elevator_direction

a case statement over this variable should not require or even allow a default field

    case $elevator_direction {
        'up' => { ... do up stuff ... }
       ' down' => { ... do down stuff ...

**Comment by ardrigh** [comment]
Surely this is answered by the README?

> Puppet Lint tests Puppet code against the recommended Puppet language style guide. Puppet Lint validates only code style; it does not validate syntax. To test syntax, use Puppet's puppet parser validate command.

puppet-lint doesn't understand data typing to my knowledge. I have run into similar questions before about how to test data typing.

**Comment by rodjek** [comment]
That's correct, puppet-lint doesn't currently have any knowledge of the data type being stored in the variable. It's possible that it could be added in a limited form in the future though it would still be limited to variables defined within the current manifest.
